subxt/src/backend/rpc/reconnecting_rpc_client.rs

```diff
@@ -0,0 +1,267 @@
+    fn default() -> Self {
+        Self {
+            max_request_size: 10 * 1024 * 1024,
+            max_response_size: 10 * 1024 * 1024,
+            retry_policy: ExponentialBackoff::from_millis(10).max_delay(Duration::from_secs(60)),
+            max_redirections: 5,
+            id_kind: IdKind::Number,
+            max_log_len: 1024,
+            max_concurrent_requests: 1024,
+            request_timeout: Duration::from_secs(60),
+            connection_timeout: Duration::from_secs(10),
+            #[cfg(feature = "native")]
+            ping_config: Some(PingConfig::new()),
+            #[cfg(feature = "native")]
+            headers: HeaderMap::new(),
+        }
+    }
+}
+
+impl Builder<ExponentialBackoff> {
+    /// Create a new builder.
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl<P> Builder<P>
+where
+    P: Iterator<Item = Duration> + Send + Sync + 'static + Clone,
+{
+    /// Configure the min response size a for websocket message.
+    ///
+    /// Default: 10MB
+    pub fn max_request_size(mut self, max: u32) -> Self {
+        self.max_request_size = max;
+        self
+    }
+
+    /// Configure the max response size a for websocket message.
+    ///
+    /// Default: 10MB
+    pub fn max_response_size(mut self, max: u32) -> Self {
+        self.max_response_size = max;
+        self
+    }
+
+    /// Set the max number of redirections to perform until a connection is regarded as failed.
+    ///
+    /// Default: 5
+    pub fn max_redirections(mut self, redirect: u32) -> Self {
+        self.max_redirections = redirect;
+        self
+    }
+
+    /// Configure how many concurrent method calls are allowed.
+    ///
+    /// Default: 1024
+    pub fn max_concurrent_requests(mut self, max: u32) -> Self {
+        self.max_concurrent_requests = max;
+        self
+    }
+
+    /// Configure how long until a method call is regarded as failed.
+    ///
+    /// Default: 1 minute
+    pub fn request_timeout(mut self, timeout: Duration) -> Self {
+        self.request_timeout = timeout;
+        self
+    }
+
+    /// Set connection timeout for the WebSocket handshake
+    ///
+    /// Default: 10 seconds
+    pub fn connection_timeout(mut self, timeout: Duration) -> Self {
+        self.connection_timeout = timeout;
+        self
+    }
+
+    /// Configure the data type of the request object ID
+    ///
+    /// Default: number
+    pub fn id_format(mut self, kind: IdKind) -> Self {
+        self.id_kind = kind;
+        self
+    }
+
+    /// Set maximum length for logging calls and responses.
+    /// Logs bigger than this limit will be truncated.
+    ///
+    /// Default: 1024
+    pub fn set_max_logging_length(mut self, max: u32) -> Self {
+        self.max_log_len = max;
+        self
+    }
+
+    /// Configure which retry policy to use.
+    ///
+    /// Default: Exponential backoff 10ms
+    pub fn retry_policy<T>(self, retry_policy: T) -> Builder<T> {
```

Comment:
So to check, `T` here is the same as `P` as in it's expected to be `Iterator<Item = Duration> + Send + Sync + 'static + Clone`?

Reply:
yes perhaps best to add these bounds here